modules/xmpp/FileSharing.ts

```diff
@@ -0,0 +1,108 @@
+import { XMPPEvents } from '../../service/xmpp/XMPPEvents';
+
+import ChatRoom from './ChatRoom';
+import XMPP from './xmpp';
+
+export const IDENTITY_TYPE = 'file-sharing';
+
+/**
+ * The FileSharing logic.
+ */
+export default class FileSharing {
+    private _mainRoom: ChatRoom;
+    private _xmpp: XMPP;
+
+    /**
+     * Constructs File sharing manager for a room.
+     *
+     * @param {ChatRoom} room the main room.
+     */
+    constructor(room: ChatRoom) {
+        this._mainRoom = room;
+        this._xmpp = room.xmpp;
+
+        this._handleMessages = this._handleMessages.bind(this);
+        this._mainRoom.xmpp.addListener(XMPPEvents.FILE_SHARING_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Stops listening for events.
+     */
+    dispose() {
+        this._mainRoom.xmpp.removeListener(XMPPEvents.FILE_SHARING_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Whether AV moderation is supported on backend.
+     *
+     * @returns {boolean} whether AV moderation is supported on backend.
+     */
+    isSupported() {
+        return Boolean(this._xmpp.fileSharingComponentAddress);
+    }
+
+    /**
+     * Returns the file sharing identity type (service name).
+     *
+     * @returns {string} the file sharing service name.
+     */
+    getIdentityType() {
+        return IDENTITY_TYPE;
+    }
+
+    addFile(metadata: any) {
```

Comment:
Do you reckon it would be useful to have an API to add multiple files in a single shot?

Reply:
Nope, we upload all files one by one: 
https://github.com/jitsi/jitsi-meet/blob/b0472e46774b6f70efa9bd98259766ac91ab6073/react/features/file-sharing/middleware.web.ts#L56

And when the upload returns 200 we add it to the list of files and broadcast the change to the other participants.